src/coreclr/src/vm/ipcstreamfactory.cpp

```diff
@@ -325,6 +325,11 @@ IpcStream *IpcStreamFactory::GetNextAvailableStream(ErrorCallback callback)
                         if (pStream == nullptr) // only use first signaled stream; will get others on subsequent calls
                         {
                             pStream = ((DiagnosticPort*)(rgIpcPollHandles[i].pUserData))->GetConnectedStream(callback);
+                            if (pStream == nullptr)
+                            {
+                                fSawError = true;
+                                break;
```

Comment:
nit: I don't this `break` is actually necessary since we set s_currentPort to `nullptr` in line 354 below anyway?